modules/default/updatenotification/update_helper.js

```diff
@@ -0,0 +1,162 @@
+const Exec = require("child_process").exec;
+const Spawn = require("child_process").spawn;
+const commandExists = require("command-exists");
+const Log = require("logger");
+
+class Updater {
+	constructor(config, callback) {
+		this.updates = config.updates;
+		this.timeout = config.updateTimeout;
+		this.autoRestart = config.updateAutorestart;
+		this.callback = callback;
+		this.moduleList = {};
+		this.updating = false;
+		this.usePM2 = false;
+		this.PM2 = null;
+		this.version = global.version;
+		this.root_path = global.root_path;
+		Log.info("updatenotification: Updater Class Loaded!");
+	}
+
+	add(modules) {
+		modules.forEach(async (module) => {
+			if (module.behind > 0) {
+				if (this.moduleList[module.module] === undefined) {
+					this.moduleList[module.module] = {};
+					this.moduleList[module.module].name = module.module;
+					this.moduleList[module.module].updateCommand = await this.applyCommand(module.module);
+					this.moduleList[module.module].neverUpdated = true;
+				}
+				if (this.moduleList[module.module].neverUpdated) {
+					if (!this.updating) {
+						if (!this.moduleList[module.module].updateCommand) {
+							this.updating = false;
+						} else {
+							this.updateProcess(this.moduleList[module.module]);
+							this.updating = true;
+						}
+					}
+					this.moduleList[module.module].neverUpdated = false;
+				}
+			} else if (module.behind === 0) {
+				if (this.moduleList[module.module] !== undefined) delete this.moduleList[module.module];
+			}
+			//Log.info("updatenotification Result:", this.moduleList[module.module])
```

Comment:
Uncomment and make a Log.debug out of it

Reply:
you have right :)

Reply:
Done